src/profiler.cpp

```diff
@@ -1680,17 +1682,27 @@ void Profiler::dumpOtlp(Writer& out, Arguments& args) {
         protobuf_mark_t sample_mark = otlp_buffer.startMessage(Profile::sample, 1);
         otlp_buffer.field(Sample::locations_start_index, frames_seen);
         otlp_buffer.field(Sample::locations_length, trace->num_frames);
-        protobuf_mark_t sample_value_mark = otlp_buffer.startMessage(Sample::value, 1);
-        otlp_buffer.putVarInt(cts->samples);
-        otlp_buffer.putVarInt(cts->counter);
-        otlp_buffer.commitMessage(sample_value_mark);
-        otlp_buffer.commitMessage(sample_mark);
 
+        u32 thread_name_idx = 0;
         for (int j = 0; j < trace->num_frames; j++) {
+            if (trace->frames[j].bci == BCI_THREAD_ID) {
+                thread_name_idx = thread_names.indexOf(fn.name(trace->frames[j]));
```

Comment:
`FrameName` returns a decorated name. Shall OTLP record the orignal name instead?

Reply:
We have [Java Thread IDs](https://github.com/async-profiler/async-profiler/blob/5454c9bf7f3628c2510b832de3676a51669b4bd7/src/profiler.h#L60), even though async-profiler uses OS tid as primary thread ID.